hack/.ci/prepare_release

```diff
@@ -25,34 +25,50 @@ apk add --no-cache \
     ca-certificates \
     make \
     bash \
-    go \
     git \
-    musl-dev \
     curl \
     openssl \
     tar \
     gzip \
-    gcc \
     sed
 
-GOLANG_VERSION="$(sed -rn 's/FROM (eu\.gcr\.io\/gardener-project\/3rd\/golang|golang):([^ ]+).*/\2/p' < "$repo_root_dir/Dockerfile")"
+# create virtual package with the dev tools
+echo "Installing dev tools"
+apk add --no-cache --virtual .build-deps \
+    gcc \
+    go \
+    musl-dev
 
-export \
-    GOROOT="$(go env GOROOT)" \
-    GOOS="$(go env GOOS)" \
-    GOARCH="$(go env GOARCH)" \
-    GOHOSTOS="$(go env GOHOSTOS)" \
-    GOHOSTARCH="$(go env GOHOSTARCH)"
+GOLANG_VERSION="$(sed -rn 's/FROM (eu\.gcr\.io\/gardener-project\/3rd\/golang|golang):([^ ]+).*/\2/p' < "$repo_root_dir/Dockerfile")"
 
-echo "Downloading go $GOLANG_VERSION"
+# alpine is using musl-libc instead of glibc, therefore we cannot use the avilable pre-built binaries from golang, but have to build them ourselves from source.
+# refs:
+#   - https://stackoverflow.com/a/45987284
+#   - https://github.com/docker-library/golang/blob/f300e60ca19c3b98cfcf01ca112af2ac10104320/1.16/alpine3.14/Dockerfile
```

Comment:
Hm, is it the actual reason? I thought that it is because we cannot specify the exact version to install.

Currently with the base image that is being used:

```
# apk add go=1.16.3
ERROR: unable to select packages:
  go-1.16.5-r0:
    breaks: world[go=1.16.3]
```

Reply:
I doubt any OS provider (in this case alpine) provides all golang versions. So we for sure cannot rely on the package registry maintained by the OS vendor and we want to install the required version by ourselves. And this another valid reason that I was not thinking about. I will add it.

Initially, I wanted to download the pre-built binaries from https://golang.org/dl/go1.16.6.linux-amd64.tar.gz, but those binaries are not compatible with alpine and actually there is no official binaries for alpine delivered by golang, therefore we build from source.